components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -421,4 +430,47 @@ public static boolean isUserPortalURL(org.wso2.carbon.identity.recovery.model.Pr
         return false;
     }
 
+    /**
+     * Check if the exception contains a password pattern violation message and act accordingly
+     *
+     * @param exception An UserStoreException
+     * @throws IdentityRecoveryClientException If exception's message contains a password pattern violation message
+     */
+    public static void checkPasswordPatternViolation(UserStoreException exception, int tenantId)
+            throws IdentityRecoveryClientException {
+
+        if (StringUtils.isNotBlank(exception.getMessage()) && StringUtils.indexOfAny(exception.getMessage(),
+                pwdPatternViolations) >= 0) {
+            UserStoreManager userStoreManager = getUserStoreManager(tenantId);
+            RealmConfiguration realmConfig = ((org.wso2.carbon.user.core.UserStoreManager)userStoreManager)
+                    .getRealmConfiguration();
+
+            String errorMessage = String.format(UserCoreErrorConstants.ErrorMessages.ERROR_CODE_INVALID_PASSWORD
+                            .getMessage(),
+                    realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_JAVA_REG_EX));
+
+            throw IdentityException.error(IdentityRecoveryClientException.class,
+                    UserCoreErrorConstants.ErrorMessages.ERROR_CODE_INVALID_PASSWORD.getCode(), errorMessage, exception);
+        }
+    }
+
+    /**
+     * Get UserStoreManager by tenantId
+     *
+     * @param tenantId tenantId
+     * @return userStoreManager UserStoreManager of the given tenant
+     * @throws IdentityRecoveryClientException If fails
+     */
+    public static UserStoreManager getUserStoreManager(int tenantId) throws IdentityRecoveryClientException {
```

Comment:
Do we need to make this method as public?

Reply:
Made the method private as it is currently used only inside the class. However, other classes may use this method to reduce code duplication.